### ISSUE TITLE
shoal client needs to indicate whether it changed the config file

### ISSUE THREAD

**Issue by rptaylor** [hep-gc/shoal#116]
If the shoal client changes the CVMFS configuration file, it will have to execute "cvmfs_config reload" to apply the change (we previously believed the automatic proxy reloading mechanism would accomplish this, but that only resets to the beginning of the list of proxies, but doesn't reload the proxies from the config file. This was previously working somehow, not sure what caused the change in behaviour).

It should not do cvmfs_config reload if the file hasn't changed, because this causes some overhead (pausing CVMFS access). Maybe the answer is just to run shoal-client less often ...

**Comment by mhpx** [comment]
having it only do the cvmfs_reload only when changes occur shouldn't be difficult.

**Comment by rptaylor** [comment]
We should think about how to do it, e.g. if shoal-client returns 0 for "success" meaning "I ran successfully and changed the config file", then you can do
```shoal-client && cvmfs_config reload```
in a cron job. There could be a different return code for "I ran successfully but didn't update the file".
The return codes should be documented in the script.

**Comment by rptaylor** [comment]
If we go that way, we'll have to check that the script gives suitable exit codes in all cases. Currently there are some ways it will fail without returning a non-zero exit code.

**Comment by rptaylor** [comment]
I contacted CVMFS developers, and they sound interested in providing an interface to update the proxy list, without having to do a config reload. That will be much much better.
So, this means that A) the original point of this ticket won't be an issue anymore (don't need to indicate whether config file was changed) and B) the shoal client will have to be updated to use the new command, e.g. "cvmfs_config proxy set" when it's available.
